src/containers/Layout/NotificationDrawerWrapper.tsx

```diff
@@ -1,42 +1,97 @@
 import {
+  Button,
   NotificationDrawer,
   NotificationDrawerBody,
   NotificationDrawerHeader,
   NotificationDrawerList,
   NotificationDrawerListItem,
   NotificationDrawerListItemBody,
-  NotificationDrawerListItemHeader
+  NotificationDrawerListItemHeader,
+  ButtonVariant
 } from '@patternfly/react-core';
-import React, { useContext } from 'react';
+import React, { useContext, useEffect, useState } from "react";
 import { AppContext } from '../../context/context';
+import { TimesIcon, TimesCircleIcon } from "@patternfly/react-icons";
+import { NotificationActionTypes } from "../../context/actions/types";
+import ChrisIntegration from "../../services/chris_integration";
+import { useHistory } from "react-router-dom";
+import { AnalysisTypes } from '../../context/actions/types';
 
 interface NotificationDrawerWrapperProps {
   onClose: () => void;
 }
 
 const NotificationDrawerWrapper: React.FC<NotificationDrawerWrapperProps> = ({ onClose }) => {
-  const { state: { notifications } } = useContext(AppContext);
+  const history = useHistory();
+  const [disabledClearAll, setDisabledClearAll] = useState(true);
+  const { state: { notifications }, dispatch } = useContext(AppContext);
+
+  const onNotificationRemoval = (index: number) => {
+    dispatch({
+      type: NotificationActionTypes.REMOVE,
+      payload: {
+        index
+      }
+    });
+  }
+
+  const onNotificationClear = () => {
+    dispatch({
+      type: NotificationActionTypes.CLEAR
+    });
+  }
+
+  useEffect(() => {
+    setDisabledClearAll(notifications.length === 0)
+  }, [notifications.length]);
+
+  const viewImg = async (index: number, id?: number) => {
+    if (id) {
+      const plugin = await ChrisIntegration.fetchPluginInstanceFromId(id);
+      const series = await ChrisIntegration.fetchResults(plugin);
+      const dcmImage = await ChrisIntegration.getDcmImageDetailByFilePathName(plugin?.data?.title);
```

Comment:
Due to optional chaining, the `getDcmImageDetailByFilePathName` parameter can be `undefined` which could cause an error since `getDcmImageDetailByFilePathName` only accepts strings.

Please, wrap an if statement around it as well as in `getDcmImageDetailByFilePathName` check that the passed value is not an empty string.

Reply:
Addressed here: https://github.com/darwinai/covidnet_ui/pull/48/commits/99f085fb3baed2cf31cb5ba21494dde270e1a3f8